src/main/java/com/projecturanus/betterp2p/network/ServerRenameP2PTunnel.kt

```diff
@@ -0,0 +1,36 @@
+package com.projecturanus.betterp2p.network
+import appeng.api.networking.IGrid
+import appeng.api.networking.IGridHost
+import appeng.parts.p2p.PartP2PTunnel
+import appeng.parts.p2p.PartP2PTunnelME
+import com.projecturanus.betterp2p.util.p2p.getEnumIndex
+import cpw.mods.fml.common.network.simpleimpl.IMessage
+import cpw.mods.fml.common.network.simpleimpl.IMessageHandler
+import cpw.mods.fml.common.network.simpleimpl.MessageContext
+import net.minecraft.tileentity.TileEntity
+import net.minecraft.world.World
+import net.minecraftforge.common.DimensionManager
+import net.minecraftforge.common.util.ForgeDirection
+fun getPart(grid: IGrid,message: C2SP2PTunnelInfo): PartP2PTunnel<*>? {
+    val listP2P: List<PartP2PTunnel<*>>? = grid.getMachines( PartP2PTunnelME::class.java)?.map { it.machine as PartP2PTunnel<*> }?.toList()
+    for (part in listP2P!!){
+        var a =part.side.getEnumIndex()
+        if(part.side.getEnumIndex() == message.info.side){ return part}
+    }
+    return null
+}
+
+class ServerRenameP2PTunnel : IMessageHandler<C2SP2PTunnelInfo, IMessage?> {
+    override fun onMessage(message: C2SP2PTunnelInfo, ctx: MessageContext): IMessage? {
+        val world: World =  DimensionManager.getWorld(message.info.world)
```

Comment:
i think you should also send the `S2CRefreshInfo` back to client side, so rename change can be updated in gui

Reply:
send name before is already updated name in the gui

Reply:
![image](https://user-images.githubusercontent.com/60341015/189523386-d5589f0e-ca79-4376-bc29-0c525a7624db.png)
the p2ps with the same frq do not change their name after renaming

Reply:
eeeeeeee  I think there is something wrong with my solution,  I will close this pr